### ISSUE TITLE
Official way of distribution

### ISSUE THREAD

**Issue by leiserfg** [KhronosGroup/Vulkan-ValidationLayers#6183]
Sorry if this is not the best place to ask (but I have no idea which one is it) so feel free to redirect me.
I'm interested in helping to update the version of vulkan distributed in nixos but I've encountered that this module depends on commits of other modules instead of on released versions. Is there an official way to gather all the components so that they are all considered "stable"?

**Comment by juan-lunarg** [comment]
> https://github.com/NixOS/nixpkgs/pull/243780

Looking at your recent PR. Number 1 issue I can point out is to wait until we make an SDK release. We just fixed a recent issue where we were pointing to a fork of glslang where we shouldn't have been.

Issues like that get ironed out before an SDK release.

> Is there an official way to gather all the components so that they are all considered "stable"?

See the `README.md`

```
Version Tagging Scheme

Updates to the Vulkan-ValidationLayers repository which correspond to a new Vulkan specification release are tagged using the following format: v<version> (e.g., v1.1.96).

Note: Marked version releases have undergone thorough testing but do not imply the same quality level as SDK tags. SDK tags follow the sdk-<version>.<patch> format (e.g., sdk-1.1.92.0).

This scheme was adopted following the 1.1.96 Vulkan specification release.
```

During SDK releases we create SDK tags/branches and then we use those tags/branches for SDK creation.

We then update a file in our SDK process called config.json which we update with the appropriate commits/tags.

```
        "Vulkan-Headers": {
            "url": "https://github.com/KhronosGroup/Vulkan-Headers.git",
            "branch": "sdk-1.3.250",
            "tag": "sdk-1.3.250.1",
            "configs": [ "Release" ]
        },
```

This applies for all KhronosGroup repos.

**Comment by leiserfg** [comment]
Thanks a lot for the details and your time.

**Comment by leiserfg** [comment]
Where can one find that config.json? Maybe using that file is the best way of updating the components downstream.

**Comment by juan-lunarg** [comment]
> I found the one at https://vulkan.lunarg.com/sdk/home#linux

That would be the best source to use.

**Comment by juan-lunarg** [comment]
> I found the one at https://vulkan.lunarg.com/sdk/home#linux but I suppose it's somewhere in github.

The SDK creation process is technically done on the Khronos Group GitLab (https://gitlab.khronos.org/vulkan/).

It's not hosted on GitHub. I don't fully understand all the reasons behind that decision (Happened before my time). So using the `config.json` provided by https://vulkan.lunarg.com/sdk/home#linux is the best you can do.

**Comment by juan-lunarg** [comment]
I ended up having this explain to me a bit further.

If you look in our `SDK release notes` and find `SDK Components` you will see the commits/tags we use for each component:
https://vulkan.lunarg.com/doc/sdk/latest/linux/release_notes.html

The `SDK Components` matches up with `config.json`.

It also contains `developer notes` and `known issues`

TLDR: config.json + release_notes documentation is all you need
